src/docker/build.rs

```diff
@@ -0,0 +1,125 @@
+use std::env;
+use std::process::Command;
+use std::str::FromStr;
+
+use super::engine::Engine;
+use crate::errors::*;
+use crate::shell::Verbosity;
+
+#[derive(Debug, PartialEq, Eq, Clone, Copy)]
+pub enum Progress {
+    Plain,
+    Auto,
+    Tty,
+}
+
+impl FromStr for Progress {
+    type Err = eyre::ErrReport;
+
+    fn from_str(progress: &str) -> Result<Self> {
+        Ok(match progress {
+            "plain" => Progress::Plain,
+            "auto" => Progress::Auto,
+            "tty" => Progress::Tty,
+            s => eyre::bail!("unexpect progress type: expected plain, auto, or tty and got {s}"),
+        })
+    }
+}
+
+impl From<Progress> for &str {
+    fn from(progress: Progress) -> Self {
+        match progress {
+            Progress::Plain => "plain",
+            Progress::Auto => "auto",
+            Progress::Tty => "tty",
+        }
+    }
+}
+
+pub trait BuildCommandExt {
+    fn build(&mut self) -> &mut Self;
+    fn progress(&mut self, progress: Option<Progress>) -> Result<&mut Self>;
+    fn verbose(&mut self, verbosity: Verbosity) -> &mut Self;
+    fn scan_suggest(&mut self, suggest: bool) -> &mut Self;
+    fn cross_labels(&mut self, target: &str, platform: &str) -> &mut Self;
+}
```

Comment:
I'm not sure I see the benefit of doing this this way.

Reply:
Have another suggestion for de-duplicating some of the code so we don't redundantly add cross labels and build commands?

Reply:
I misunderstood the intent of this. My concern was with making it hard to decouple xtask and custom build in cross, but as it is now it's fine.